src/wacks4shop/shift4shop.py

```diff
@@ -1,29 +1,113 @@
 import logging
 import os
+from datetime import datetime
 from typing import Optional
 
+import requests
+
+from wacksbywarby.constants import SHIFT4SHOP_ORDER_DATE_FORMAT, SHIFT4SHOP_TIME_FORMAT
 from wacksbywarby.models import Sale
 
 logger = logging.getLogger("shift4shop")
 
+BASE_API = "https://apirest.3dcart.com/3dCartWebAPI/v1"
+"""
+Use all order statuses except for 7, not completed
+From @rchhay:
+1 = New
+2 = Processing
+3 = Partial
+4 = Shipped
+5 = Cancel
+6 = Hold
+7 = Not Completed
+8 = Custom 1 (I think this is for gift card)
+9 = Custom 2 (not being used i think)
+10 = Custom 3 (also not used)
+"""
+ORDER_STATUSES = [1, 2, 3, 4, 5, 6, 8, 9, 10]
+
 
 class Shift4Shop:
     def __init__(self, debug=False) -> None:
         self.key = os.getenv("SHIFT4SHOP_PRIVATE_KEY")
         self.shop_token = os.getenv("SHIFT4SHOP_SHOP_TOKEN")
         self.debug = debug
+        self.headers = {
+            "SecureURL": "wicksbywerby.com",
+            "PrivateKey": self.key,
+            "Token": self.shop_token,
+        }
+
+    def _request_orders(self, params):
+        return requests.get(f"{BASE_API}/Orders", params=params, headers=self.headers)
+
+    def _request_product_details(self, catalog_id):
+        return requests.get(f"{BASE_API}/Products/{catalog_id}", headers=self.headers)
+
+    def request_all_products(self):
+        """
+        A util function that is useful for figuring out the catalog IDs of werbies in
+        shift4shop
+        """
+        response = requests.get(
+            f"{BASE_API}/Products/", headers=self.headers, params={"limit": 100}
+        )
+        products = response.json()
+        for product in products:
+            print(product["SKUInfo"]["CatalogID"], product["SKUInfo"]["Name"], "\n")
```

Comment:
do you still need this print?

Reply:
this was a helper func to list out each product's ID in shift4shop (unlike etsy, shift4shop does not include the listing's id in the URL).

I refactored to lift it into a first class citizen with proper logging and documented it in the README https://github.com/allisonking/wacks-by-warby/pull/11/commits/94bbabc96005a635f71b30d1db7e1c8e6bbefbc6

check out `python3 -m wacks4shop.ls` 😎